src/main/scala/cromwell/util/TryUtil.scala

```diff
@@ -16,4 +19,54 @@ object TryUtil {
 
   def stringifyFailures[T](possibleFailures: Traversable[Try[T]]): Traversable[String] =
     possibleFailures.collect { case failure: Failure[T] => stringifyFailure(failure) }
+
+  private def defaultSuccessFunction(a: Any): Boolean = true
+
+  /**
+   * Runs a block of code (`fn`) `retries` number of times until it succeeds.
+   * It will wait `pollingInterval` amount of time between retry attempts and
+   * The `pollingBackOffFactor` is for exponentially backing off the `pollingInterval`
+   * on subsequent retries.  The `pollingInterval` shall not exceed `maxPollingInterval`
+   *
+   * Returns a Try[T] where T is the return value of `fn`, the function to be retried.
+   * If the return value is Success[T] then at least one retry succeeded.
+   *
+   * The isSuccess function is optional but if provided, then isSuccess(fn) must be true
+   * or it will trigger another retry.  if isSuccess is omitted, the only way the fn can
+   * fail is if it throws an exception.
+   *
+   * Use `retries` value of -1 to indicate retrying indefinitely.
```

Comment:
as someone who has spent more brain-hours C-ing than anything, I :+1:  that comment. But i always used -666 for bad stuff.